fed/mosaic-mapping/src/main/java/org/eclipse/mosaic/fed/mapping/ambassador/MappingAmbassador.java

```diff
@@ -124,39 +121,37 @@ private void handleInteraction(ScenarioTrafficLightRegistration interaction) {
      */
     private void handleInteraction(ScenarioVehicleRegistration scenarioVehicle) throws InternalFederateException {
         if (framework != null) {
-            final CPrototype prototype = framework.getPrototypeByName(scenarioVehicle.getVehicleTypeId());
+            final CPrototype prototype = framework.getPrototypeByName(scenarioVehicle.getVehicleType().getName());
             if (prototype == null) {
                 log.debug(
                         "There is no such prototype \"{}\" configured. No application will be mapped for vehicle \"{}\".",
-                        scenarioVehicle.getVehicleTypeId(),
+                        scenarioVehicle.getVehicleType().getName(),
                         scenarioVehicle.getId()
                 );
                 return;
             }
-            List<String> applications = prototype.applications == null ? new ArrayList<>() : prototype.applications;
-            String group = prototype.group == null ? null : prototype.group;
+
             if (randomNumberGenerator.nextDouble() >= ObjectUtils.defaultIfNull(prototype.weight, 1.0)) {
                 log.debug(
                         "This scenario vehicle \"{}\" of prototype \"{}\" will not be equipped due to a weight condition of {}.",
-                        scenarioVehicle.getVehicleTypeId(),
+                        scenarioVehicle.getVehicleType().getName(),
                         scenarioVehicle.getId(),
                         prototype.weight
                 );
-                applications = new ArrayList<>();
-                group = null;
+                return;
             }
```

Comment:
@schwepmo I reverted your change and we now again send `VehicleRegistration` for true matches only. `ApplicationAmbassador` is now able to process `ScenarioVehicleRegistration` where it extracts the `VehicleType` and passes it to the perception index.

Reply:
@schwepmo are you fine with this specfic change?

Reply:
Yes, for me this works fine. I'm not sure if we need to adjust the `runtime.json` in MOSAIC extended as well?

Reply:
That's part of another merge request